src/platform_impl/linux/x11/window.rs

```diff
@@ -180,13 +181,41 @@ impl UnownedWindow {
         };
 
         // creating
+        let (depth, visual) = match pl_attribs.visual_infos {
+            Some(vi) => (vi.depth, vi.visual),
+            None if window_attrs.transparent == true => {
+                // Find a suitable visual
+                let mut vinfo = MaybeUninit::uninit();
+                if unsafe {
+                    (xconn.xlib.XMatchVisualInfo)(
+                        xconn.display,
+                        screen_id,
+                        32,
+                        TrueColor,
+                        vinfo.as_mut_ptr(),
+                    )
+                } != 0
+                {
```

Comment:
It might be cleaner to extract this into a variable.

```suggestion
                let vinfo_initialized = unsafe {
                    (xconn.xlib.XMatchVisualInfo)(
                        xconn.display,
                        screen_id,
                        32,
                        TrueColor,
                        vinfo.as_mut_ptr(),
                    ) != 0
                };
                if vinfo_initialized {
```

Reply:
Done